connect/api/src/main/java/org/apache/kafka/connect/source/SourceRecord.java

```diff
@@ -46,25 +46,49 @@
     private final Map<String, ?> sourcePartition;
     private final Map<String, ?> sourceOffset;
 
+    /**
+     * @deprecated use {@link SourceRecord.Builder}
+     */
+    @Deprecated
```

Comment:
Just my humble opinion, but I prefer the constructors over the builder since it makes it easy to see which fields are needed and it's generally less verbose. But there are some situations where it a Builder may be more helpful, so I think it makes sense to offer that as an alternative.